src/modules/awake/Awake/Core/APIHelper.cs

```diff
@@ -43,21 +43,21 @@ static APIHelper()
             _tokenSource = new CancellationTokenSource();
         }
 
-        public static void SetConsoleControlHandler(ConsoleEventHandler handler, bool addHandler)
+        internal static void SetConsoleControlHandler(PHANDLER_ROUTINE handler, bool addHandler)
         {
-            NativeMethods.SetConsoleCtrlHandler(handler, addHandler);
+            PInvoke.SetConsoleCtrlHandler(handler, addHandler);
         }
 
         public static void AllocateConsole()
         {
             _log.Debug("Bootstrapping the console allocation routine.");
-            NativeMethods.AllocConsole();
+            PInvoke.AllocConsole();
             _log.Debug($"Console allocation result: {Marshal.GetLastWin32Error()}");
 
-            var outputFilePointer = NativeMethods.CreateFile("CONOUT$", GenericRead | GenericWrite, FileShare.Write, IntPtr.Zero, FileMode.OpenOrCreate, 0, IntPtr.Zero);
+            var outputFilePointer = PInvoke.CreateFile("CONOUT$", FILE_ACCESS_FLAGS.FILE_GENERIC_READ | FILE_ACCESS_FLAGS.FILE_GENERIC_WRITE, FILE_SHARE_MODE.FILE_SHARE_WRITE, null, FILE_CREATION_DISPOSITION.CREATE_NEW, 0, null);
```

Comment:
Is OpenOrCreate the same as CREATE_NEW?

Reply:
shouldn't matter. As if file exists, OpenOrCreate will rewrite the file, not seek to the end and append

Reply:
But will `FILE_CREATION_DISPOSITION.CREATE_NEW` fail if the file is there already?

Reply:
Yeah, before it was a bug, in a sense that we were passing dotnet enum value to a native method. it was the same as passing `FILE_CREATION_DISPOSITION.OPEN_ALWAYS` (also equals 0x4). However, the next line is logging `CONOUT creation result`, so the intention was to create something, that's why I've replaced it with `FILE_CREATION_DISPOSITION.CREATE_NEW`.

Reply:
That's how strong typing from CsWin32 helps find bugs 🙂

Reply:
after reading [the docs](https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-createfilea#consoles) we should actually use `OPEN_EXISTING` for devices:
![image](https://user-images.githubusercontent.com/1828123/189948027-90486172-417e-44b5-b2df-9ae772d0656d.png)

Anyway, that should be a safe change, since that code only runs if we launch awake directly and not from the runner, which is our only officially supported flow.